modules/twinkleconfig.js

```diff
@@ -1396,88 +1396,79 @@ Twinkle.config.listDialog.display = function twinkleconfigListDialogDisplay(e) {
 	dialog.setTitle(curpref.label);
 	dialog.setScriptName('Twinkle preferences');
 
-	var dialogcontent = document.createElement('div');
-	var dlgtable = document.createElement('table');
-	dlgtable.className = 'wikitable';
-	dlgtable.style.margin = '1.4em 1em';
-	dlgtable.style.width = 'auto';
-
-	var dlgtbody = document.createElement('tbody');
-
-	// header row
-	var dlgtr = document.createElement('tr');
-	// top-left cell
-	var dlgth = document.createElement('th');
-	dlgth.style.width = '5%';
-	dlgtr.appendChild(dlgth);
-	// value column header
-	dlgth = document.createElement('th');
-	dlgth.style.width = '35%';
-	dlgth.textContent = curpref.customListValueTitle ? curpref.customListValueTitle : 'Value';
-	dlgtr.appendChild(dlgth);
-	// label column header
-	dlgth = document.createElement('th');
-	dlgth.style.width = '60%';
-	dlgth.textContent = curpref.customListLabelTitle ? curpref.customListLabelTitle : 'Label';
-	dlgtr.appendChild(dlgth);
-	dlgtbody.appendChild(dlgtr);
+	var $dlgtbody;
+
+	dialog.setContent(
+		$('<div>').append(
+			$('<table>')
+				.addClass('wikitable')
+				.css({
+					margin: '1.4em 1em',
+					width: 'auto'
+				})
+				.append(
+					$dlgtbody = $('<tbody>').append(
+						// header row
+						$('<tr>').append(
+							$('<th>') // top-left cell
+								.css('width', '35%'),
+							$('<th>') // value column header
+								.css('width', '35%')
+								.text(curpref.customListValueTitle ? curpref.customListValueTitle : 'Value'),
+							$('<th>') // label column header
+								.css('width', '60%')
+								.text(curpref.customListLabelTitle ? curpref.customListLabelTitle : 'Label')
+						)
+					),
+					$('<tfoot>').append(
+						$('<tr>').append(
+							$('<td>')
+								.attr('colspan', '3')
+								.append(
+									$('<button>')
+										.text('Add')
+										.css('minWidth', '8em')
+										.attr('type', 'button')
+										.on('click', function () {
+											Twinkle.config.listDialog.addRow($dlgtbody);
+										})
+								)
+						)
+					)
+				),
+			$('<button>')
+				.text('Save changes')
+				.attr('type', 'submit') // so Morebits.simpleWindow puts the button in the button pane
+				.on('click', function () {
+					Twinkle.config.listDialog.save($prefbutton, $dlgtbody);
```

Comment:
Minor, but as long as we're updating one place `Twinkle.config.listDialog.save` and `Twinkle.config.listDialog.reset` are called, mightn't we likewise change their parameters to `$button, $tbody`?  Could be some minor cleanup in each.

Reply:
done